sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -530,6 +543,10 @@ public boolean advance() throws IOException {
         // we consume message without autoAck (we want to do the ack ourselves)
         GetResponse delivery = channel.basicGet(queueName, false);
         if (delivery == null) {
+          current = null;
```

Comment:
Notable change since the last round of reviews: The documentation for `Source`'s `getCurrent`, `getCurrentRecordId` and `getCurrentTimestamp` include the following:

>      * @throws NoSuchElementException if the reader is at the beginning of the input and {@link
>     *     #start} or {@link #advance} wasn't called, or if the last {@link #start} or {@link
>     *     #advance} returned {@code false}.

This states that if `advance()` returns `false`, subsequent calls to these should throw `NoSuchElementException`. As such, I'm explicitly setting these to `null`.